cpp/src/gandiva/precompiled/time.cc

```diff
@@ -706,10 +707,14 @@ const char* castVARCHAR_timestamp_int64(int64 context, timestamp in, int64 lengt
   char char_buffer[char_buffer_length];
 
   // yyyy-MM-dd hh:mm:ss.sss
-  snprintf(char_buffer, char_buffer_length,
-           "%04" PRId64 "-%02" PRId64 "-%02" PRId64 " %02" PRId64 ":%02" PRId64
-           ":%02" PRId64 ".%03" PRId64,
-           year, month, day, hour, minute, second, millis);
+  int res = snprintf(char_buffer, char_buffer_length,
+                     "%04" PRId64 "-%02" PRId64 "-%02" PRId64 " %02" PRId64 ":%02" PRId64
+                     ":%02" PRId64 ".%03" PRId64,
+                     year, month, day, hour, minute, second, millis);
+  if (res < 0) {
+    DCHECK(false);  // should be unreachable.
```

Comment:
It could definitively fail, why not set the error with `gdv_fn_context_set_error_msg` like the other error return path?

Reply:
I stopped using DCHECK, and follow the standard error process with `gdv_fn_context_set_error_msg`.